control

```diff
@@ -153,64 +153,58 @@ class ControlINI(ConfigParser.SafeConfigParser, Singleton):
     Representation of control settings
     """
 
-    # Default relative locations for control settings files
-    CONTROL_INI_DEFAULT = "config_defaults/control.ini"
-    CONTROL_INI_CUSTOM = "config_custom/control.ini"
-
-    # Mapping of option-name to section name (for generating defaults)
-    OPT_SEC_MAP = {'include': 'Control',
-                   'exclude': 'Control',
-                   'subthings': 'Control',
-                   'pretests': 'Control',
-                   'subtests': 'Control',
-                   'intratests': 'Control',
-                   'posttests': 'Control',
-                   'url': 'Bugzilla',
-                   'username': 'Bugzilla',
-                   'password': 'Bugzilla',
-                   'excluded': 'Bugzilla',
-                   'key_field': 'Bugzilla',
-                   'key_match': 'Bugzilla',}
-
-    # Absolute base path where all other relative paths reside
+    # Absolute base path to prefix onto below
     control_path = os.path.dirname(job.control)
 
+    # Default relative locations, prefixed by above
+    control_ini_default = os.path.join(control_path,
+                                       "config_defaults/control.ini")
+    control_ini_custom = os.path.join(control_path,
+                                      "config_custom/control.ini")
+
     # Default location where write() writes to if no file given
     write_path = job.resultdir
 
     # Token that signals not to execute tests
     NOEXECTOK = '!!!'
 
     def __init__(self):
-        # Help catch missing options in defaults but not in OPT_SEC_MAP
-        # by making them None instead of ''
-        super(ControlINI, self).__init__(allow_no_value=True)
+        super(ControlINI, self).__init__(self.defaults(), allow_no_value=True)
         self.optionxform = str  # support case-sensitive options
-        for option, section in self.OPT_SEC_MAP.iteritems():
-            try:
-                self.add_section(section)
-            except ConfigParser.DuplicateSectionError:
-                pass  # already existing section
-            # Empty string is the default value
-            self.set(section, option, '')
-        # These option values default to the option name, look up section
-        for key in ('pretests', 'subtests', 'intratests', 'posttests'):
-            self.set(self.OPT_SEC_MAP[key], key, key)
+
+    def defaults(self):
```

Comment:
Never mind; looks like you fixed that in a force push.

I'm going to back away until this is stable.